connector/signaltometricsconnector/config/config.go

```diff
@@ -0,0 +1,67 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package config // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/signaltometricsconnector/config"
+
+import "fmt"
+
+// Config for the connector. Each configuration field describes the metrics
+// to produce from a specific signal.
+type Config struct {
+	Spans      []MetricInfo `mapstructure:"spans"`
+	Datapoints []MetricInfo `mapstructure:"datapoints"`
+	Logs       []MetricInfo `mapstructure:"logs"`
+}
+
+func (c *Config) Validate() error {
+	if len(c.Spans) == 0 && len(c.Datapoints) == 0 && len(c.Logs) == 0 {
+		return fmt.Errorf("no configuration provided, at least one should be specified")
+	}
+	return nil
+}
+
+// MetricInfo defines the structure of the metric produced by the connector.
+type MetricInfo struct {
+	Name        string `mapstructure:"name"`
+	Description string `mapstructure:"description"`
+	// Unit, if not-empty, will set the unit associated with the metric.
+	// See: https://github.com/open-telemetry/opentelemetry-collector/blob/b06236cc794982916cc956f20828b3e18eb33264/pdata/pmetric/generated_metric.go#L72-L81
+	Unit string `mapstructure:"unit"`
+	// IncludeResourceAttributes is a list of resource attributes that
+	// needs to be included in the generated metric. If no resource
+	// attribute is included in the list then all attributes are included.
+	// Metric data streams MUST obey single-writer. The component produces
+	// metrics from non-metric signals as well as alters resource attributes
+	// from the source signals. To keep the single-writer valid, the
+	// component adds collector information as resource attribute with the
+	// component name as the prefix of the resource attributes.
+	IncludeResourceAttributes []Attribute `mapstructure:"include_resource_attributes"`
+	Attributes                []Attribute `mapstructure:"attributes"`
+	// Conditions are a set of OTTL condtions which are ORd. Data is
```

Comment:
Could you elaborate on what this `ORd` stands for? Apologies it's something I should know :)

Reply:
Ah, no worries - should have done a better job explaining that. What I mean is that all OTTL conditions are ORd together and the metric will be produced IFF the resulting value is `true`. Example:

```yaml
    name: test.metric
    conditions:
      - resource.attributes["attribute1"] != nil
      - resource.attributes["attribute2"] != nil
```

The above definition will produce `test.metric` metric only if the resource attributes have either `attribute1` OR `attribute2` present.

Will also add some more documentation there.